### PR TITLE
Allow nuclear operatives to upload their mind into spawned cyborgs

### PR DESCRIPTION
**Pull request by datlo**
Nuclear operatives now get a prompt when spawning a syndicate cyborg asking if they want to play as the op or the cyborg. 

image

If they choose to play as the borg, ghosts will be asked if they wish to play as the operative. If none accept, the borg will not be spawned and the spawner will not be used.

🆑
add: Nuclear operatives can now choose to take over a cyborg when spawning one, with a ghost playing the nuclear operative.
/🆑

### CODE REVIEW

**Comment by Shadeykins** [comment]
Oh! That's super neat.

You're full of good ideas.

**Comment by KasparoVy** [comment]
Not sure if we're ever gonna see non-op borgs after this one 🤔 good idea

**Comment by Fox-McCloud** [comment]
Yeah---I admit I'm a bit skeptical on this.

**Comment by Shadeykins** [comment]
> 
> 
> Not sure if we're gonna see too many ghostspawn borgs after this one 🤔 I wonder how it'll end up

The operative themselves would get offered up to ghosts.

**Comment by KasparoVy** [comment]
Yeah but what I'm trying to say is that now I figure not too many ghosts will get the chance to be borgs, mostly nukies who switch over

But only time and data will tell

**Comment by EvadableMoxie** [comment]
Even if everyone just chooses to play the borg, I don't see it as an issue unless no ghosts take the Operative which I think is probably unlikely.  Players who readied up with nukie activated and won the antag role should have first dibs over ghosts getting a role, shouldn't they?

**Comment by Denthamos** [comment]
Huh, well this is neat.  I like it, as it will let the nukies have some more flexibility if they need a cyborg for their plans but no ghosts wish to take part.

File: code/modules/antagonists/_common/antag_spawner.dm [hunk 1]
```diff
@@ -34,20 +35,42 @@
...
+	if(upload_mind == "Nuclear Operative")
+		checking = TRUE
+		to_chat(user, "<span class='notice'>The device is now checking for possible borgs.</span>")
+		var/list/borg_candidates = pollCandidates("Do you want to play as a Syndicate [borg_to_spawn] borg?", ROLE_OPERATIVE, 1)
+		if(borg_candidates.len > 0 && !used)
+			checking = FALSE
+			used = TRUE
+			var/mob/M = pick(borg_candidates)
+			var/client/C = M.client
+			spawn_antag(C, get_turf(src.loc), "syndieborg")
```
Comment (variableundefined) [inline code/modules/antagonists/_common/antag_spawner.dm]: src?

File: code/modules/antagonists/_common/antag_spawner.dm [hunk 2]
```diff
@@ -34,20 +35,42 @@
...
+		checking = TRUE
+		to_chat(user, "<span class='notice'>You attempt to upload your consciousness into a new syndicate cyborg.</span>")
+		var/list/nuclear_candidates = pollCandidates("Do you want to play as the Syndicate Nuclear Operative [user.real_name]?", ROLE_OPERATIVE, 1)
+		if(nuclear_candidates.len > 0 && !used)
+			checking = FALSE
+			used = TRUE
+			var/mob/M = pick(nuclear_candidates)
+			var/client/nukeop_client = M.client
+			var/client/borg_client = user.client
+			spawn_antag(borg_client, get_turf(src.loc), "syndieborg")
```
Comment (variableundefined) [inline code/modules/antagonists/_common/antag_spawner.dm]: src

**Comment by datlo** [comment]
removed implied srcs

**Pull request merged**